test/z_general_permission_manager_fuzzer.js

```diff
@@ -0,0 +1,1263 @@
+    const decimals = 18;
+    const contact = "team@polymath.network";
+    const delegateDetails = "Hello I am legit delegate";
+    const STVRParameters = ["bool", "uint256", "bool"];
+
+    // Module key
+    const delegateManagerKey = 1;
+    const transferManagerKey = 2;
+    const stoKey = 3;
+
+    // Initial fee for ticker registry and security token registry
+    const initRegFee = web3.utils.toWei("250");
+
+    // CountTransferManager details
+	const holderCount = 2; // Maximum number of token holders
+	let bytesSTO = encodeModuleCall(["uint256"], [holderCount]);
+
+	let _details = "details holding for test";
+    let testRepeat = 20;
+
+	// permission manager fuzz test
+	let perms = ['ADMIN','WHITELIST', 'FLAGS', 'TRANSFER_APPROVAL'];
+	let totalPerms = perms.length;
+
+    before(async () => {
+        // Accounts setup
+        account_polymath = accounts[0];
+        account_issuer = accounts[1];
+
+        token_owner = account_issuer;
+        token_owner_pk = pk.account_1;
+
+        account_investor1 = accounts[8];
+        account_investor2 = accounts[9];
+        account_investor3 = accounts[5];
+        account_investor4 = accounts[6];
+        account_delegate = accounts[7];
+        // account_delegate2 = accounts[6];
+
+
+        // Step 1: Deploy the genral PM ecosystem
+        let instances = await setUpPolymathNetwork(account_polymath, token_owner);
+
+        [
+            I_PolymathRegistry,
+            I_PolyToken,
+            I_FeatureRegistry,
+            I_ModuleRegistry,
+            I_ModuleRegistryProxy,
+            I_MRProxied,
+            I_GeneralTransferManagerFactory,
+            I_STFactory,
+            I_SecurityTokenRegistry,
+            I_SecurityTokenRegistryProxy,
+            I_STRProxied
+        ] = instances;
+
+        // STEP 5: Deploy the GeneralDelegateManagerFactory
+        [I_GeneralPermissionManagerFactory] = await deployGPMAndVerifyed(account_polymath, I_MRProxied, I_PolyToken.address, 0);
+        // STEP 6: Deploy the GeneralDelegateManagerFactory
+        [P_GeneralPermissionManagerFactory] = await deployGPMAndVerifyed(account_polymath, I_MRProxied, I_PolyToken.address, web3.utils.toWei("500"));
+
+	    // Deploy Modules
+        [I_CountTransferManagerFactory] = await deployCountTMAndVerifyed(account_polymath, I_MRProxied, I_PolyToken.address, 0);
+
+        [I_VolumeRestrictionTransferManagerFactory] = await deployLockupVolumeRTMAndVerified(account_polymath, I_MRProxied, I_PolyToken.address, 0);
+
+        [I_PercentageTransferManagerFactory] = await deployPercentageTMAndVerified(account_polymath, I_MRProxied, I_PolyToken.address, 0);
+
+        [I_SingleTradeVolumeRestrictionManagerFactory] = await deploySingleTradeVolumeRMAndVerified(account_polymath, I_MRProxied, I_PolyToken.address, 0);
+        [P_SingleTradeVolumeRestrictionManagerFactory] = await deploySingleTradeVolumeRMAndVerified(account_polymath, I_MRProxied, I_PolyToken.address, web3.utils.toWei("500"));
+
```

Comment:
Remove it

Reply:
@comeonbuddy Please can you take a look at this.

Reply:
done